File: backend/app/schemas/course.py
"""Sprint 7 course schemas (catalog + enrollment projections)."""

from datetime import datetime

from pydantic import BaseModel, Field


class CourseCatalogItem(BaseModel):
    slug: str
    title: str
    description: str | None
    sort_order: int


class CourseCatalogOut(BaseModel):
    items: list[CourseCatalogItem]


class EnrollmentOut(BaseModel):
    """Returned in admin user detail and as part of /api/courses."""

    course_slug: str
    course_title: str
    status: str = Field(pattern=r"^(active|paused|completed)$")
    enrolled_at: datetime


class MyCourseItem(BaseModel):
    """A course the authenticated learner is enrolled in."""

    slug: str
    title: str
    description: str | None
    status: str = Field(pattern=r"^(active|paused|completed)$")


class MyCoursesOut(BaseModel):
    items: list[MyCourseItem]

File: frontend/src/types/course.ts
/**
 * Sprint 7 course DTOs.
 *
 * Mirrors `backend/app/schemas/course.py`. Field names and casing match
 * the server payload so JSON.parse output is assignable without a
 * mapping layer.
 */

export interface CourseCatalogItem {
  slug: string;
  title: string;
  description: string | null;
  sort_order: number;
}

export interface CourseCatalogResponse {
  items: CourseCatalogItem[];
}

export type EnrollmentStatus = 'active' | 'paused' | 'completed';

export interface MyCourseItem {
  slug: string;
  title: string;
  description: string | null;
  status: EnrollmentStatus;
}

export interface MyCoursesResponse {
  items: MyCourseItem[];
}

/**
 * Admin-facing enrollment projection — matches
 * `backend/app/schemas/course.py::EnrollmentOut`.
 */
export interface EnrollmentOut {
  course_slug: string;
  course_title: string;
  status: EnrollmentStatus;
  enrolled_at: string;
}
